src/encoded/static/components/inputs/file.js

```diff
@@ -2,6 +2,39 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 
+/**
+ * Depending on the filename extension, adjust the type and base64-encoded contents from the
+ * defaults that FileReader assigns. Return unchanged results for any file types that don't need
+ * special treatment.
+ * @param {object} rFile Object from <input type="file" />
+ * @param {string} readerResult FileReader `result` property.
+ *
+ * @return {object} -
+ *   type - mime type for file
+ *   href - base64-encoded contents of file
+ */
+const adjustFileValues = (rFile, readerResult) => {
```

Comment:
@forresttanaka  Is this a valid alternative:
const adjustFileValues = (rFile, readerResult) => {
    // https://stackoverflow.com/questions/190852/how-can-i-get-file-extensions-with-javascript/1203361#answer-12900504
 const extension = rFile.name.slice((Math.max(0, rFile.name.lastIndexOf('.')) || Infinity) + 1);
 
const isFileTypeAs =  extension === 'as’;
 const type = isFileTypeAs ?  'text/autosql' :  rFile.type;
 const href = isFileTypeAs ? readerResult.replace(/^(data:).+?(;.*)$/, '$1text/autosql$2') : readerResult

    return {
        type,
        href,
    };
}

Reply:
My spider sense are tingling when I see you only check for .as extension coded but not other types.

Reply:
What would you do with the other types? Only .as files need anything special done with them, as documented here and in the ticket.

Reply:
As the only function of this line of code is to get a file extension, and the discussion is documented in the link, and for us this doesn’t happen in a tight loop so performance isn’t a concern, I wasn't willing to spend any more time on it.